probe_scraper/probe_expiry_alert.py

```diff
@@ -0,0 +1,106 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from collections import defaultdict
+import argparse
+import datetime
+
+import requests
+
+from probe_scraper import emailer
+from probe_scraper.scrapers import release_calendar
+
+FROM_EMAIL = "telemetry-alerts@mozilla.com"
+DEFAULT_TO_EMAIL = "dev-telemetry-alerts@mozilla.com"
+EMAIL_TIME_BEFORE = datetime.timedelta(weeks=2)
+PROBE_INFO_BASE_URL = "https://probeinfo.telemetry.mozilla.org/"
+
+
+def find_expiring_probes(target_date, probes, release_dates):
+    """
+    Find probes expiring on the target date using output of the probe info service
+
+    Returns dict of form:
+    {
+      channel: {
+        histogram_name: [notification_emails...]
+      }
+    }
+    """
+    expiring_histograms_by_channel = defaultdict(dict)
+    for probe in probes.values():
+        for channel, details in probe["history"].items():
+            details = details[0]
+            expiry_version = details["expiry_version"]
+            if expiry_version == "never":
+                continue
+
+            expiry_date = release_dates[channel].get(expiry_version)
+            if expiry_date == target_date:
+                expiring_histograms_by_channel[channel][probe["name"]] = (
+                    details.get("notification_emails", []) + [DEFAULT_TO_EMAIL])
+
+    return expiring_histograms_by_channel
+
+
+def send_emails_for_expiring_probes(target_date, expiring_histograms_by_channel, dryrun=True):
+    histograms_by_email = defaultdict(lambda: defaultdict(list))
+
+    for channel, histograms in expiring_histograms_by_channel.items():
+        for name, emails in histograms.items():
+            for email in emails:
+                histograms_by_email[email][channel].append(name)
+
+    email_body_format_string = """
```

Comment:
I think we can clean this message up a bit to make it more actionable. Something like:

```
The following Firefox probes will expire in the next {expire_days} days or have already expired.

What to do about this:
1. If the metric is no longer needed, ignore this warning. You can remove it from `histograms.json`[link] or `scalars.yaml`[link], but then it will not be available to view in telemetry.mozilla.org (however, it will still be queryable).
2. If the metric is still required, resubmit a data review [1] and extend its expiration date.
If you have any problems, please ask for help on the #fx-metrics Slack channel. We'll give you a hand.

Your Friendly, Neighborhood Telemetry Team
[1] https://wiki.mozilla.org/Firefox/Data_Collection
This is an automated message sent from probe-scraper.  See https://github.com/mozilla/probe-scraper for details.
```

cc @chutten

Reply:
For Case 1 I'd like to de-emphasize that "do nothing" is an acceptable choice. We want people to clean up their probes (and in the case of the version increase simulator resulting in sheriffs filing bugs like [this one](https://bugzilla.mozilla.org/show_bug.cgi?id=1607737), this might be working against what we've previous said), so maybe something like.

```
1. If one, some, or all of the metrics are no longer needed, please remove them from their definitions files (Histograms.json, Scalars.yaml, Events.yaml).
2. If one, some, or all of the metrics are still required, please submit a Data Collection Review and patch to extend their expiry.
```

Two things that definitely need to be watched are 
1. Histogram.json/Scalars.yaml are Capital Case
2. It's an expiry, not an expiry date. (Expiry dates are for Glean)

Otherwise: all good.